lib/failure_app.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# Custom failure app to handle warden errors
+class FailureApp < Devise::FailureApp
+  def http_auth_body
+    {
+      attribute: 'email',
```

Comment:
This is only added because the frontend expects an attribute key in the error responses to fetch the right error message. I am using `email` because it applies to both confirmation errors and authentication errors. But it could be anything and we only have to define a mapping in the error messages used on the frontend. Let me know if we should update it.

Reply:
I like this a lot. :)